### PR TITLE
Implement `command + enter` for apply modification

### PR DESCRIPTION
**Pull request by akanex115**
# Background
https://github.com/ajktown/wordnote/issues/202

## TODOs
- [x] The dialog reads the input `command + enter` and run `modify`

## Checklist (Developers)
- [x] `Draft` is set for this PR
- [x] `Title` is checked
- [x] `Background` is filled
- [x] `TODOs` are filled
- [x] `Assignee` is set
- [x] `Labels` are set
- [x] `Project` is created & linked, if multiple PRs are associated to this PR
- [x] `development` is linked if related issue exists
- [x] `yarn inspect` is run
- [x] `TODOs` are handled and checked
- [x] Final Operation Check is done
- [x] Make this PR as an open PR

## Checklist (Reviewers)
- [x] Check if there are any other missing TODOs that are not yet listed
- [x] Review Code
- [x] Every item on the checklist has been addressed accordingly
- [x] If `development` is associated to this PR, you must check if every TODOs are handled

### CODE REVIEW

**Review by mlajkim — COMMENTED** [review]
Nice job. but please do the following instead:
1. Modify `src/hooks/use-key-press.hook.ts` to handle multiple keys
  - Make sure that it takes the first argument as the function when pressed
  - and the rest is all the key presses
2. Use the modified hook to implement this

**Comment by mlajkim** [comment]
👍 
image

File: src/global.interface.ts [hunk 1]
```diff
@@ -37,6 +37,8 @@ export type GlobalKeyboardEventKey =
   | 'ArrowLeft'
   | 'ArrowUp'
   | 'ArrowDown'
+  | 'Meta'
+  | 'Control'
```
Comment (mlajkim) [inline src/global.interface.ts:41]: Can you write a simple comment like this:
```suggestion
  | 'Meta' // ⌘ in Apple Ecosystem
  | 'Control'
```

File: src/components/molecule_new_word_box/index.tsx [hunk 2]
```diff
@@ -38,8 +38,8 @@ const NewWordBox: FC = () => {
     else setWritingMode(true)
   }, [isWritingMode, setWritingMode, onDynamicFocus])
 
-  useKeyPress(`Enter`, onHitEnter)
-  useKeyPress(`Escape`, onClickPostWordWritingModeClose)
+  useKeyPress(onHitEnter, `Enter`)
```
Comment (mlajkim) [inline src/components/molecule_new_word_box/index.tsx:41]: This probably requires to change onHitEnter not to enter, if a dialog is open

File: src/hooks/use-key-press.hook.ts [hunk 3]
```diff
@@ -20,5 +24,5 @@ export const useKeyPress = (
     return () => {
       document.removeEventListener(`keydown`, onKeyDown)
     }
-  }, [keyName, onKeyPress])
+  }, [onKeyPress, ...keyNames])
```
Comment (mlajkim) [inline src/hooks/use-key-press.hook.ts:27]: Yeah, you must fix the errors here:
image

**Review by mlajkim — CHANGES_REQUESTED** [review]
first review

File: src/components/molecule_new_word_box/index.tsx (hunk 2 above)
Comment (mlajkim) [reply src/components/molecule_new_word_box/index.tsx:41]: (should modify onHitEnter logic, where it should stop itself if a dialog is opened (like edit dialog)

File: src/hooks/use-key-press.hook.ts (hunk 3 above)
Comment (akanex115) [reply src/hooks/use-key-press.hook.ts]: Handled in [bd174bd](https://github.com/ajktown/wordnote/pull/210/commits/bd174bd481b2b2a340a90dfd9680111fe24080eb)

File: src/components/molecule_new_word_box/index.tsx (hunk 2 above)
Comment (akanex115) [reply src/components/molecule_new_word_box/index.tsx:41]: I will make an issue for this and handle in a different PR

File: src/components/atom_word_card_confirm_modify_button/index.tsx [hunk 4]
```diff
@@ -21,6 +22,21 @@ const WordCardConfirmModifyButton: FC<Props> = ({ wordId }) => {
...
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if ((event.metaKey || event.ctrlKey) && event.key === `Enter`) {
+        handleApplyCache()
+      }
+    }
+    document.addEventListener(`keydown`, handleKeyDown)
+    return () => {
+      document.removeEventListener(`keydown`, handleKeyDown)
+    }
+  })
```
Comment (akanex115) [inline src/components/atom_word_card_confirm_modify_button/index.tsx:35]:
```suggestion
```

File: src/components/atom_word_card_confirm_modify_button/index.tsx [hunk 5]
```diff
@@ -21,6 +22,10 @@ const WordCardConfirmModifyButton: FC<Props> = ({ wordId }) => {
     `exampleLink`,
   )
 
+
+  useKeyPress(handleApplyCache, `Meta`, `Enter`)
+  useKeyPress(handleApplyCache, `Control`, `Enter`)
```
Comment (akanex115) [inline src/components/atom_word_card_confirm_modify_button/index.tsx:27]:
```suggestion
  useKeyPress(handleApplyCache, `Meta`, `Enter`) // for mac
  useKeyPress(handleApplyCache, `Control`, `Enter`) // for windows
```

File: src/hooks/use-key-press.hook.ts (hunk 3 above)
Comment (akanex115) [reply src/hooks/use-key-press.hook.ts]: Handled in [a1f5eee](https://github.com/ajktown/wordnote/pull/210/commits/a1f5eee512f21a4c5e4dc680b645ecf5931157c3)

**Review by mlajkim — APPROVED** [review]
OK

**Pull request merged**